### PR TITLE
III-4410 Create repository for news article table

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added value object for `NewsArticle` and a simple `NewsArticles` collection
- Added repository `NewsArticleRepository` and a concrete DBAL implementation `DBALNewsArticleRepository`

---
Ticket: https://jira.uitdatabank.be/browse/III-4410

### CODE REVIEW

File: src/Curators/NewsArticleNotFoundException.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Curators;
+
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use Exception;
+
+final class NewsArticleNotFoundException extends Exception
```
Comment (jdrieghe) [inline src/Curators/NewsArticleNotFoundException.php:10]: Is there a reason why we still postfix exception classes with `...Exception`?
I think `NewsArticleNotFound()` is clear enough.

File: src/Curators/DBALNewsArticleRepository.php [hunk 2]
```diff
@@ -51,6 +51,39 @@ public function getAll(): NewsArticles
         return $this->createNewsArticles($newsArticleRows);
     }
 
+    public function create(NewsArticle $newsArticle): void
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
```
Comment (jdrieghe) [inline src/Curators/DBALNewsArticleRepository.php:56]: Isn't there an `->insert()` method on DBAL connections that can make this implementation a lot shorter?

File: src/Curators/DBALNewsArticleRepository.php [hunk 3]
```diff
@@ -84,6 +84,17 @@ public function create(NewsArticle $newsArticle): void
             ->execute();
     }
 
+    public function delete(UUID $id): void
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
```
Comment (jdrieghe) [inline src/Curators/DBALNewsArticleRepository.php:118]: Same here, I think there is a `->delete()` shorthand

File: src/Curators/NewsArticleNotFoundException.php (hunk 1 above)
Comment (LucWollants) [reply src/Curators/NewsArticleNotFoundException.php]: Good point, I changed it to `NewsArticleNotFound`

File: src/Curators/DBALNewsArticleRepository.php (hunk 2 above)
Comment (LucWollants) [reply src/Curators/DBALNewsArticleRepository.php:56]: There is indeed a shorthand `insert` on the `Connection` class but that is not escaped and are not safe for user input. So I always stick with the `QueryBuilder` methods and only use the shorthand ones inside tests.

See https://github.com/doctrine/dbal/blob/v2.9.3/lib/Doctrine/DBAL/Connection.php#L745

File: src/Curators/DBALNewsArticleRepository.php (hunk 3 above)
Comment (LucWollants) [reply src/Curators/DBALNewsArticleRepository.php:118]: Same issue with the `delete` method of `Connection` https://github.com/doctrine/dbal/blob/v2.9.3/lib/Doctrine/DBAL/Connection.php#L636

**Pull request merged**